### ISSUE TITLE
story isn't getting linted (eslint)

### ISSUE THREAD

**Issue by jimmy-e** [storybookjs/storybook#7178]
### Bug or support request summary

I am trying to reconfigure my babel building my storybook no longer works. I am suspicious that it has something to do with my babel configs? I am able to build and run my React SPA successfully. The failure only occurs when trying to run storybook.

### Steps to reproduce

See [this repo](https://github.com/jimmy-e/storybook-lint-issue).

Clone it and run the following commands:

`yarn`

`yarn storybook`

### You will get the following error:

```
WARNING in ./src/sampleComponent.stories.jsx
Module build failed (from ./node_modules/eslint-loader/index.js):
Module failed because of a eslint error.

/Users/jemery/dev/storybook-lint-issue/src/sampleComponent.stories.jsx
   2:1    error  Expected indentation of 0 spaces but found 2                                                        indent
   2:3    error  Expected 1 empty line after require statement not followed by another require                       import/newline-after-import
   2:10   error  Exporting mutable 'var' binding, use 'const' instead                                                import/no-mutable-exports
   2:10   error  Unexpected var, use let or const instead                                                            no-var
   2:14   error  Use object destructuring                                                                            prefer-destructuring
   2:32   error  Unexpected require()                                                                                global-require
   2:32   error  '@storybook/addon-storysource' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   2:32   error  `@storybook/addon-storysource` import should occur after import of `./sampleComponent`              import/order
   3:1    error  Expected indentation of 0 spaces but found 2                                                        indent
   3:10   error  Exporting mutable 'var' binding, use 'const' instead                                                import/no-mutable-exports
   3:10   error  Unexpected var, use let or const instead                                                            no-var
   3:14   error  Unexpected dangling '_' in '__STORY__'                                                              no-underscore-dangle
   4:1    error  Expected indentation of 0 spaces but found 2                                                        indent
   4:10   error  Exporting mutable 'var' binding, use 'const' instead                                                import/no-mutable-exports
   4:10   error  Unexpected var, use let or const instead                                                            no-var
   4:14   error  Unexpected dangling '_' in '__ADDS_MAP__'                                                           no-underscore-dangle
   4:29   error  A space is required after '{'                                                                       object-curly-spacing
   4:30   error  Strings must use singlequote                                                                        quotes
   4:67   error  Missing space before value for key 'sample-component--sample-component'                             key-spacing
   4:67   error  A space is required after '{'                                                                       object-curly-spacing
   4:68   error  Unnecessarily quoted property 'startLoc' found                                                      quote-props
   4:68   error  Strings must use singlequote                                                                        quotes
   4:79   error  Missing space before value for key 'startLoc'                                                       key-spacing
   4:79   error  A space is required after '{'                                                                       object-curly-spacing
   4:80   error  Unnecessarily quoted property 'col' found                                                           quote-props
   4:80   error  Strings must use singlequote                                                                        quotes
   4:86   error  Missing space before value for key 'col'                                                            key-spacing
   4:88   error  A space is required after ','                                                                       comma-spacing
   4:89   error  Unnecessarily quoted property 'line' found                                                          quote-props
   4:89   error  Strings must use singlequote                                                                        quotes
   4:96   error  Missing space before value for key 'line'                                                           key-spacing
   4:97   error  A space is required before '}'                                                                      object-curly-spacing
   4:98   error  A space is required after ','                                                                       comma-spacing
   4:99   error  Unnecessarily quoted property 'endLoc' found                                                        quote-props
   4:99   error  Strings must use singlequote                                                                        quotes
   4:108  error  Missing space before value for key 'endLoc'                                                         key-spacing
   4:108  error  A space is required after '{'                                                                       object-curly-spacing
   4:109  error  Unnecessarily quoted property 'col' found                                                           quote-props
   4:109  error  Strings must use singlequote                                                                        quotes
   4:115  error  Missing space before value for key 'col'                                                            key-spacing
   4:117  error  A space is required after ','                                                                       comma-spacing
   4:118  error  Unnecessarily quoted property 'line' found                                                          quote-props
   4:118  error  Strings must use singlequote                                                                        quotes
   4:125  error  Missing space before value for key 'line'                                                           key-spacing
   4:126  error  A space is required before '}'                                                                      object-curly-spacing
   4:127  error  A space is required before '}'                                                                      object-curly-spacing
   4:128  error  A space is required before '}'                                                                      object-curly-spacing
   5:1    error  Trailing spaces not allowed                                                                         no-trailing-spaces
   6:1    error  Expected indentation of 0 spaces but found 2                                                        indent
   6:3    error  Import in body of module; reorder to top                                                            import/first
   7:1    error  Import in body of module; reorder to top                                                            import/first
   7:1    error  '@storybook/react' should be listed in the project's dependencies, not devDependencies              import/no-extraneous-dependencies
   8:1    error  Import in body of module; reorder to top                                                            import/first
  14:1    error  Too many blank lines at the end of file. Max of 0 allowed                                           no-multiple-empty-lines
  15:1    error  Trailing spaces not allowed                                                                         no-trailing-spaces
  15:3    error  Newline required at end of file but not found                                                       eol-last

✖ 56 problems (56 errors, 0 warnings)
  45 errors and 0 warnings potentially fixable with the `--fix` option.

 @ ./src sync \.stories\.jsx$ ./sampleComponent.stories.jsx
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/config.js (webpack)-hot-middleware/client.js?reload=true
Child HtmlWebpackCompiler:
```
### Relevant Scripts

**package.json:**
```
{
  "name": "bei-bei",
  "version": "1.7.0",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook",
    "build-storybook": "build-storybook -c .storybook -o .out",
    "remove-dist": "rm -rf dist",
    "copy-index": "cp src/index.html dist/index.html",
    "devWebpack": "webpack --config webpack.dev.js",
    "prodWebpack": "webpack --config webpack.prod.js",
    "start": "webpack-dev-server --inline --hot --history-api-fallback --config webpack.dev.js",
    "build:dev": "npm-run-all remove-dist devWebpack copy-index",
    "build:internal": "INTERNAL=true npm-run-all remove-dist prodWebpack copy-index",
    "build:prod": "npm-run-all remove-dist prodWebpack copy-index"
  },
  "license": "UNLICENSED",
  "devDependencies": {
    "@babel/core": "^7.4.5",
    "@babel/preset-env": "^7.4.5",
    "@babel/preset-react": "^7.0.0",
    "@fortawesome/fontawesome-pro": "^5.2.0",
    "@storybook/addon-a11y": "^5.1.3",
    "@storybook/addon-actions": "^5.1.3",
    "@storybook/addon-backgrounds": "^5.1.3",
    "@storybook/addon-console": "^1.1.0",
    "@storybook/addon-info": "^5.1.3",
    "@storybook/addon-knobs": "^5.1.3",
    "@storybook/addon-storysource": "^5.1.3",
    "@storybook/addon-viewport": "^5.1.3",
    "@storybook/addons": "^5.1.3",
    "@storybook/react": "^5.1.3",
    "babel-cli": "7.0.0-beta.3",
    "babel-eslint": "7",
    "babel-jest": "^24.5.0",
    "babel-loader": "^8.0.6",
    "babel-plugin-module-resolver": "^3.2.0",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-2": "^6.24.1",
    "browserslist": "^4.3.1",
    "css-loader": "^0.28.7",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "eslint": "^5.16",
    "eslint-config-airbnb": "^17.1.0",
    "eslint-import-resolver-webpack": "^0.11.1",
    "eslint-loader": "^2.1.2",
    "eslint-plugin-import": "^2.17.2",
    "eslint-plugin-jest": "^22.5.1",
    "eslint-plugin-jsx-a11y": "^6.2.1",
    "eslint-plugin-react": "^7.12.4",
    "file-loader": "^1.1.11",
    "identity-obj-proxy": "^3.0.0",
    "image-webpack-loader": "^4.3.1",
    "jest": "^23.1.0",
    "jest-enzyme": "^4.0.0",
    "mini-css-extract-plugin": "^0.4.1",
    "node-sass": "^4.12.0",
    "npm-run-all": "^4.1.3",
    "postcss-loader": "^3.0.0",
    "redux-mock-store": "^1.5.3",
    "sass-loader": "^7.1.0",
    "style-loader": "^0.23.1",
    "url-loader": "^1.1.1",
    "webpack": "^4.16.5",
    "webpack-cli": "^3.1.0",
    "webpack-dev-server": "^3.1.5",
    "webpack-merge": "^4.1.4"
  },
  "dependencies": {
    "@fortawesome/fontawesome": "^1.1.8",
    "@fortawesome/fontawesome-svg-core": "^1.2.0-14",
    "@fortawesome/free-brands-svg-icons": "^5.1.0-11",
    "@fortawesome/free-regular-svg-icons": "^5.1.0-11",
    "@fortawesome/free-solid-svg-icons": "^5.1.0-11",
    "@fortawesome/pro-light-svg-icons": "^5.1.0-11",
    "@fortawesome/pro-regular-svg-icons": "^5.1.0-11",
    "@fortawesome/pro-solid-svg-icons": "^5.1.0-11",
    "@fortawesome/react-fontawesome": "0.1.0-11",
    "@sentry/browser": "^4.3.2",
    "@trendmicro/react-toggle-switch": "^0.5.7",
    "autosize": "^4.0.2",
    "axios": "^0.18.0",
    "babel-plugin-transform-builtin-extend": "^1.1.2",
    "babel-polyfill": "^6.26.0",
    "connected-react-router": "^5.0.1",
    "font-awesome": "^4.7.0",
    "history": "^4.7.2",
    "js-cookie": "^2.2.0",
    "less-loader": "^4.0.5",
    "libphonenumber-js": "^0.4.42",
    "lodash": "^4.17.4",
    "mdi-react": "^4.0.0",
    "moment": "^2.19.1",
    "prop-types": "^15.6.0",
    "query-string": "^5.0.1",
    "rc-time-picker": "^3.1.0",
    "react": "^16.4.0",
    "react-animations": "^1.0.0",
    "react-autosuggest": "^9.3.4",
    "react-circular-progressbar": "^0.8.0",
    "react-datepicker": "^0.59.0",
    "react-dom": "^16.4.0",
    "react-highcharts": "^16.0.2",
    "react-list": "^0.8.8",
    "react-redux": "^5.1.1",
    "react-router": "^4.3.1",
    "react-router-dom": "^4.2.2",
    "react-select": "^1.0.0-rc.10",
    "react-transition-group": "^1.2.0",
    "redux": "^3.7.2",
    "redux-thunk": "^2.2.0",
    "reselect": "^4.0.0",
    "styled-components": "3.2.3",
    "twilio-client": "^1.6.5"
  }
}

```

**.babelrc:**
```
{
  "presets": ["@babel/env", "@babel/react"]
}

```

**.eslintrc:**
```
module.exports = {
  parser: 'babel-eslint',
  extends: 'airbnb',
  plugins: ['jest'],
  env: {
    'jest/globals': true
  },
  rules: {
    'react/destructuring-assignment': [false, 'always', { 'ignoreClassFields': true }], // ToDo: Get rid of this rule:
    'react/require-default-props': 0, // ToDo: get rid of this rule.
    'jsx-a11y/label-has-associated-control': 0, // ToDo: get rid of this rule.
    'jsx-a11y/label-has-for': 0, // ToDo: get rid of this rule.
    'react/no-array-index-key': 0, // ToDo: Review: permanent rule?
    'import/prefer-default-export': 0, // ToDo: Review: permanent rule?
    'import/no-extraneous-dependencies': [
      'error',
      {
        'devDependencies': ['**/*.stories.js'],
      },
    ],
  },
  settings: {
    'import/extensions': [
      '.js',
      '.jsx'
    ],
    'import/resolver': {
      webpack: {
        config: 'webpack.common.js',
      }
    }
  },
  globals: {
    'document': true,
    'window': true,
  },
};

```

**storybook config.js:**
```
import React from 'react';
import { configure } from '@storybook/react';
import addonConfigs from './addons.configs';

addonConfigs();

const req = require.context('../src/', true, /\.stories\.jsx$/);

const loadStories = () => {
  req.keys().forEach(filename => req(filename));
};

configure(loadStories, module);

```

**storybook webpack:**
```
const merge = require('webpack-merge');
const common = require('../webpack.common.js');

const config = merge(common, {
  module: {
    rules: [
      {
        test: /\.stories\.jsx?$/,
        loaders: [require.resolve('@storybook/addon-storysource/loader')],
        enforce: 'pre',
      },
    ],
  },
});

module.exports = config;

```

**webpack.common.js:**
```
const path = require('path');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

const DIST_DIR = path.resolve(__dirname, 'dist');
const SRC_DIR = path.resolve(__dirname, 'src');

const config = {
  entry: [
    `${SRC_DIR}/index.jsx`,
  ],
  output: {
    path: `${DIST_DIR}/app/`,
    filename: 'bundle.js',
    publicPath: '/app/',
  },
  module: {
    rules: [
      {
        enforce: 'pre',
        test: /\.jsx?$/,
        exclude: /node_modules/,
        loader: 'eslint-loader',
        options: {
          failOnWarning: true,
          failOnError: true,
        },
      },
      {
        test: /\.jsx?$/,
        include: SRC_DIR,
        loader: 'babel-loader',
      },
      {
        test: /\.css$/,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
        ],
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: 'application.css',
    }),
  ],
  resolve: {
    extensions: ['.jsx', '.js'],
  },
};

module.exports = config;

```

**index.jsx:**
```
import 'babel-polyfill';
import React from 'react';
import { render } from 'react-dom';
import SampleComponent from './sampleComponent';

render(
  <SampleComponent />, document.getElementById('app'),
);
```

**sampleComponent.jsx:**
```
import React from 'react';

const SampleComponent = () => <h1>This is a sample component!</h1>;

export default SampleComponent;
```

**sampleComponent.stories.jsx:**
```
import React from 'react';
import { storiesOf } from '@storybook/react';
import SampleComponent from './sampleComponent';

storiesOf('Sample Component', module)
  .add('Sample Component', () => (
    <SampleComponent />
  ));
```

**Comment by shilman** [comment]
I'm getting an error trying to install your repro, but browsing the code this is what I think is happening:

1. Your common config is using the eslint-loader that checks every file as its loaded
2. Your storybook config is using the storysource loader that adds some variables into your stories file, and is enforced as "pre", meaning it runs before the eslint-loader.
3. The eslint-loader is analyzing the output of storysource, and barfing.

If this is correct, I would suggest one or more of the following:
- Run eslint by hand on your stories files just to make sure they are clear
- Exclude eslint loader for stories files
- Enforce eslint loader as "pre" and hope that it gets added before storysource (not sure how to do this)

**Comment by libetl** [comment]
Same problem as in #6974 ?

**Comment by jimmy-e** [comment]
@libetl , you are correct, this issue is actually occuring because I forgot to remove the storysource loader in my storybook webpack config. That issue should be addressed separately in issue [#6974](https://github.com/storybookjs/storybook/issues/6974).

Thus, updating the storybook webpack to the following should fix this issue:

**storybook webpack:**
```js
const merge = require('webpack-merge');
const common = require('../webpack.common.js');

const config = merge(common, {});

module.exports = config;
```
